extensions/machinepools/vsphere_machine_config.go

```diff
@@ -9,86 +9,115 @@ const (
+
+	Hostsystem string `json:"hostsystem" yaml:"hostsystem"`
+	Datacenter string `json:"datacenter" yaml:"datacenter"`
+	Datastore  string `json:"datastore" yaml:"datastore"`
+	Folder     string `json:"folder" yaml:"folder"`
+	Pool       string `json:"pool" yaml:"pool"`
+}
+
 // VsphereMachineConfig is configuration needed to create an rke-machine-config.cattle.io.vmwarevsphereconfig
 type VmwarevsphereMachineConfig struct {
-	Cfgparam              []string `json:"cfgparam" yaml:"cfgparam"`
-	CloneFrom             string   `json:"cloneFrom" yaml:"cloneFrom"`
-	CloudConfig           string   `json:"cloudConfig" yaml:"cloudConfig"`
-	Cloundinit            string   `json:"cloundinit" yaml:"cloundinit"`
-	ContentLibrary        string   `json:"contentLibrary" yaml:"contentLibrary"`
-	CPUCount              string   `json:"cpuCount" yaml:"cpuCount"`
-	CreationType          string   `json:"creationType" yaml:"creationType"`
-	CustomAttribute       []string `json:"customAttribute" yaml:"customAttribute"`
-	DataCenter            string   `json:"dataCenter" yaml:"dataCenter"`
-	DataStore             string   `json:"dataStore" yaml:"dataStore"`
-	DatastoreCluster      string   `json:"datastoreCluster" yaml:"datastoreCluster"`
-	DiskSize              string   `json:"diskSize" yaml:"diskSize"`
-	Folder                string   `json:"folder" yaml:"folder"`
-	HostSystem            string   `json:"hostSystem" yaml:"hostSystem"`
-	MemorySize            string   `json:"memorySize" yaml:"memorySize"`
-	Network               []string `json:"network" yaml:"network"`
-	OS                    string   `json:"os" yaml:"os"`
-	Password              string   `json:"password" yaml:"password"`
-	Pool                  string   `json:"pool" yaml:"pool"`
-	SSHPassword           string   `json:"sshPassword" yaml:"sshPassword"`
-	SSHPort               string   `json:"sshPort" yaml:"sshPort"`
-	SSHUser               string   `json:"sshUser" yaml:"sshUser"`
-	SSHUserGroup          string   `json:"sshUserGroup" yaml:"sshUserGroup"`
-	Tag                   []string `json:"tag" yaml:"tag"`
-	Username              string   `json:"username" yaml:"username"`
-	VappIpallocationplicy string   `json:"vappIpallocationplicy" yaml:"vappIpallocationplicy"`
-	VappIpprotocol        string   `json:"vappIpprotocol" yaml:"vappIpprotocol"`
-	VappProperty          []string `json:"vappProperty" yaml:"vappProperty"`
-	VappTransport         string   `json:"vappTransport" yaml:"vappTransport"`
-	Vcenter               string   `json:"vcenter" yaml:"vcenter"`
-	VcenterPort           string   `json:"vcenterPort" yaml:"vcenterPort"`
+	Roles
+	Boot2dockerURL          string   `json:"boot2dockerURL" yaml:"boot2dockerURL"`
+	Cfgparam                []string `json:"cfgparam" yaml:"cfgparam"`
+	CloneFrom               string   `json:"cloneFrom" yaml:"cloneFrom"`
+	CloudConfig             string   `json:"cloudConfig" yaml:"cloudConfig"`
+	Cloundinit              string   `json:"cloundinit" yaml:"cloundinit"`
```

Comment:
I know this is a bit outside the scope, but should be a small change. This is a typo that needs to be `Cloudinit` instead of `Cloundinit`.

Reply:
good catch; resolved